core/src/main/scala/kafka/zk/KafkaZkClient.scala

```diff
@@ -88,11 +88,50 @@ class KafkaZkClient private[zk] (zooKeeperClient: ZooKeeperClient, isSecure: Boo
     * @param brokerInfo payload of the broker znode
     * @return broker epoch (znode create transaction id)
     */
-  def registerBroker(brokerInfo: BrokerInfo): Long = {
+  def registerBroker(brokerInfo: BrokerInfo): BrokerEpochAndZkVersion = {
     val path = brokerInfo.path
     val stat = checkedEphemeralCreate(path, brokerInfo.toJsonBytes)
     info(s"Registered broker ${brokerInfo.broker.id} at path $path with addresses: ${brokerInfo.broker.endPoints}, czxid (broker epoch): ${stat.getCzxid}")
-    stat.getCzxid
+    BrokerEpochAndZkVersion(stat.getCzxid, stat.getVersion)
+  }
+
+  /**
+   * Unregisters the broker
+   * @param brokerInfo payload of the broker being unregistered
+   * @param brokerZkVersion version for the broker znode
+   */
+  def unregisterBroker(brokerInfo: BrokerInfo, brokerZkVersion: Int): Unit = {
+    val response = retryRequestUntilConnected(
+      DeleteRequest(brokerInfo.path, brokerZkVersion))
+
+    info(s"unregistering broker ${brokerInfo.path} version $brokerZkVersion")
+    response.resultCode match {
+      // Code.BADVERSION may indicate that another instance with the same ID has taken over.
+      // This should only happen for a new instance, so we should treat this as a success.
+      case Code.NONODE | Code.OK =>
+      case Code.BADVERSION => info(s"unregistering broker was no-op due to version mismatch")
+      case code => throw KeeperException.create(code)
+    }
+  }
+
+  /**
+   * Unregisters the broker and unregisters it as a controller if it is one.
+   * @param brokerInfo payload of the broker being unregistered
+   */
```

Comment:
param docs for **brokerZkVersion** and **expectedControllerEpochZkVersion** are missing

tbh those parameter names are self-explaining, although I would keep the docs consistent with the function above, which provides a param doc for **brokerZkVersion** as well